GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/LooseObjectStepTests.cs

```diff
@@ -25,28 +26,59 @@ public LooseObjectStepTests()
         private string PackRoot => this.Enlistment.GetPackRoot(this.fileSystem);
         private string TempPackRoot=> Path.Combine(this.PackRoot, TempPackFolder);
 
+        [SetUp]
+        public void DeleteLooseObjectsAndMovePackFiles()
+        {
+            // Delete/Move any starting loose objects and packfiles
+            this.DeleteFiles(this.GetLooseObjectFiles());
+            this.MovePackFilesToTemp();
+            Assert.AreEqual(0, this.GetLooseObjectFiles().Count);
+            Assert.AreEqual(0, this.CountPackFiles());
+        }
+
         [TestCase]
         public void RemoveLooseObjectsInPackFiles()
         {
-            // Delete any starting loose objects and verify
-            this.DeleteFiles(this.GetLooseObjectFiles());
+            // Copy and expand one pack
+            this.ExpandOneTempPack(copyPackBackToPackDirectory: true);
+            Assert.AreNotEqual(0, this.GetLooseObjectFiles().Count);
+            Assert.AreEqual(1, this.CountPackFiles());
+
+            // Cleanup should delete all loose objects, since they are in the packfile
+            this.Enlistment.LooseObjectStep();
+
             Assert.AreEqual(0, this.GetLooseObjectFiles().Count);
+            Assert.AreEqual(1, this.CountPackFiles());
+        }
 
-            // Move packfiles to temp
-            this.MovePackFilesToTemp();
+        [TestCase]
+        public void PutLooseObjectsInPackFiles()
+        {
+            // Expand one pack, and verify we have loose objects
+            this.ExpandOneTempPack(copyPackBackToPackDirectory: false);
+            int loooseObjectCount = this.GetLooseObjectFiles().Count();
+            Assert.AreNotEqual(0, loooseObjectCount);
 
-            // Expand 1 pack file and Copy it back to packs
-            this.ExpandOneTempPackAndMoveBack();
+            // This step should put the loose objects into a packfile
+            this.Enlistment.LooseObjectStep();
 
-            // Verify we have some LooseObjects
-            // These objects will also appear in the pack file that was moved back
-            Assert.AreNotEqual(0, this.GetLooseObjectFiles().Count);
+            Assert.AreEqual(loooseObjectCount, this.GetLooseObjectFiles().Count);
+            Assert.AreEqual(1, this.CountPackFiles());
+
+            // Running the step a second time should remove the loose obects and keep the pack file
+            this.Enlistment.LooseObjectStep();
+
+            Assert.AreEqual(0, this.GetLooseObjectFiles().Count);
+            Assert.AreEqual(1, this.CountPackFiles());
+        }
 
-            // Run Cleanup
+        [TestCase]
+        public void NoLooseObjectsDoesNothing()
+        {
             this.Enlistment.LooseObjectStep();

```

Comment:
I think you need to order your test cases explicitly to ensure this test runs after the previous tests. That way you know there are no loose objects. Otherwise, you'll want to delete all loose objects in this test before running the step.